app/services/provider_interface/save_condition_statuses.rb

```diff
@@ -0,0 +1,56 @@
+module ProviderInterface
+  class SaveConditionStatuses
+    include ImpersonationAuditHelper
+
+    def initialize(actor:, application_choice:, statuses_form_object:)
+      @auth = ProviderAuthorisation.new(actor: actor)
+      @application_choice = application_choice
+      @statuses_form_object = statuses_form_object
+    end
+
+    def save!
+      auth.assert_can_make_decisions!(application_choice: application_choice, course_option: application_choice.current_course_option)
+
+      audit(auth.actor) do
+        ActiveRecord::Base.transaction do
+          save_conditions_and_update_application!
+        end
```

Comment:
We are raising a `Workflow::NoTransitionAllowed` if the transition isn't allowed, should we rescue this error and add the relevant error message to handle this gracefully rather than return a 500?

Reply:
I've looked into this now, thanks.

I think we should still throw the error, but I'm adding better redirect logic so that the error won't happen unless something has actually gone wrong

Reply:
See the new logic in the controller to redirect with a message if the application is no longer pending conditions